server/src/main/java/io/crate/planner/optimizer/joinorder/JoinGraph.java

```diff
@@ -83,7 +80,7 @@
  * </pre>
  */
 public record JoinGraph(List<LogicalPlan> nodes,
-                        Map<LogicalPlan, Set<Edge>> edges,
+                        Map<LogicalPlan, List<Edge>> edges,
```

Comment:
Order of the Edges should be preserved to avoid flaky test output. Lists are also more efficient than Sets.

Reply:
No, because conceptionally in a join-tree each equi-join edge exists only once afaik.